src/adapter/src/catalog.rs

```diff
@@ -360,6 +360,11 @@ impl Catalog {
         }
         policies
     }
+
+    /// WIP
+    pub fn hack_add_ct(&mut self, id: GlobalId, entry: CatalogEntry) {
+        self.state.entry_by_id.insert(id, entry);
+    }
```

Comment:
To make sure I understand, this hack exists because Continual Tasks are the first kind of object that are self referential? Can you add the reason we have the hack as a doc comment on this method?